src/BridgedRoom.ts

```diff
@@ -954,6 +954,17 @@ export class BridgedRoom {
         // Dedupe across RTM/Event streams
         this.addRecentSlackMessage(message.ts);
 
+        if (this.SlackType === "im") {
+            const intent = await this.getIntentForRoom();
+            if (intent.userId !== ghost.intent.userId &&
+                !(await intent.matrixClient.getRoomMembers(this.matrixRoomId, undefined, ["invite", "join"]))
+                    .map(m => m.membershipFor).includes(ghost.intent.userId))
+            {
+                await intent.invite(this.matrixRoomId, ghost.matrixUserId);
```

Comment:
Should we skip inviting if we know the user is invited from the prior call, if the API tells us?

Reply:
AFAICT the API tells us by having `Intent.invite` throw an error if the user is joined already!  It also seems to be indistinguishable from a "real" error... So checking room membership in advance might be the only safe way to invite someone.

Unless you meant something else?

Reply:
Well it's just you were checking the room membership above, in `getRoomMembers`. Can't we short circuit on join?

Reply:
The tricky part is that what triggers this is seeing an inbound Slack message, which doesn't strongly correspond to a Matrix ghost having joined. So I don't think there's anything before this can be used as a hint to ignore having to send an invite.

Reply:
Okay, that makes sense to me. My point was only an optimization if there was an easy path :)